src/python/pants/core/goals/test.py

```diff
@@ -77,15 +74,26 @@ def from_fallible_process_result(
         *,
         coverage_data: Optional["CoverageData"] = None,
         xml_results: Optional[Digest] = None,
+        address_ref: str = "",
     ) -> "TestResult":
         return TestResult(
             status=Status.SUCCESS if process_result.exit_code == 0 else Status.FAILURE,
             stdout=process_result.stdout.decode(),
             stderr=process_result.stderr.decode(),
             coverage_data=coverage_data,
             xml_results=xml_results,
+            address_ref=address_ref,
         )
 
+    def level(self):
+        if self.status == Status.FAILURE:
+            return LogLevel.WARN
```

Comment:
This should probably be ERROR.

Reply:
I’m not sure. The user is expecting that things might fail, ie it’s a fallible process. 

But I can see arguments for both.

Reply:
I think that given that the pants run as a whole will fail if a test fails, this should probably be ERROR... warnings shouldn't cause the process to fail.